pkg/collector/corechecks/oracle-dbm/statements.go

```diff
@@ -222,6 +266,126 @@ type FQTPayload struct {
+	Tables   []string `json:"tables"`
+	Commands []string `json:"commands"`
+	Comments []string `json:"comments"`
+}
+
+type PlanDefinition struct {
+	Operation        string  `json:"operation,omitempty"`
+	Options          string  `json:"options,omitempty"`
+	ObjectOwner      string  `json:"object_owner,omitempty"`
+	ObjectName       string  `json:"object_name,omitempty"`
+	ObjectAlias      string  `json:"object_alias,omitempty"`
+	ObjectType       string  `json:"object_type,omitempty"`
+	PlanStepId       int64   `json:"id,omitempty"`
+	ParentId         int64   `json:"parent_id,omitempty"`
+	Depth            int64   `json:"depth,omitempty"`
+	Position         int64   `json:"position,omitempty"`
+	SearchColumns    int64   `json:"search_columns,omitempty"`
+	Cost             float64 `json:"cost,omitempty"`
+	Cardinality      float64 `json:"cardinality,omitempty"`
+	Bytes            float64 `json:"bytes,omitempty"`
+	PartitionStart   string  `json:"partition_start,omitempty"`
+	PartitionStop    string  `json:"partition_stop,omitempty"`
+	CPUCost          float64 `json:"cpu_cost,omitempty"`
+	IOCost           float64 `json:"io_cost,omitempty"`
+	TempSpace        float64 `json:"temp_space,omitempty"`
+	AccessPredicates string  `json:"access_predicates,omitempty"`
+	FilterPredicates string  `json:"filter_predicates,omitempty"`
+	Projection       string  `json:"projection,omitempty"`
+	LastStarts       uint64  `json:"actual_starts,omitempty"`
+	LastOutputRows   uint64  `json:"actual_rows,omitempty"`
+	LastCRBufferGets uint64  `json:"actual_cr_buffer_gets,omitempty"`
+	LastDiskReads    uint64  `json:"actual_disk_reads,omitempty"`
+	LastDiskWrites   uint64  `json:"actual_disk_writes,omitempty"`
+	LastElapsedTime  uint64  `json:"actual_elapsed_time,omitempty"`
+	LastMemoryUsed   uint64  `json:"actual_memory_used,omitempty"`
+	LastDegree       uint64  `json:"actual_parallel_degree,omitempty"`
+	LastTempsegSize  uint64  `json:"actual_tempseg_size,omitempty"`
+}
+
+type PlanPlanDB struct {
+	Definition []PlanDefinition `json:"definition"`
+	Signature  string           `json:"signature"`
+}
+
+type PlanDB struct {
+	Instance       string                `json:"instance,omitempty"`
+	Plan           PlanPlanDB            `json:"plan,omitempty"`
+	QuerySignature string                `json:"query_signature,omitempty"`
+	Statement      string                `json:"statement,omitempty"`
+	Metadata       PlanStatementMetadata `json:"metadata,omitempty"`
+}
+
+type PlanPayload struct {
+	Timestamp    float64    `json:"timestamp,omitempty"`
+	Host         string     `json:"host,omitempty"` // Host is the database hostname, not the agent hostname
+	AgentVersion string     `json:"ddagentversion,omitempty"`
+	Source       string     `json:"ddsource"`
+	Tags         string     `json:"ddtags,omitempty"`
+	DBMType      string     `json:"dbm_type"`
+	PlanDB       PlanDB     `json:"db"`
+	OraclePlan   OraclePlan `json:"oracle"`
+}
+
+type PlanGlobalRows struct {
```

Comment:
Unless I'm misreading, this struct (and others below) hold data for a single row. I'd suggest just naming them singular (i.e. `PlangGlobalRow`).